bpf/cpu/cpu.bpf.c

```diff
@@ -300,31 +300,37 @@ static __always_inline int walk_user_stacktrace(bpf_user_pt_regs_t *regs,
     u64 found_pc = unwind_table->rows[table_idx].pc;
     u64 found_cfa_reg = unwind_table->rows[table_idx].cfa_reg;
     u64 found_cfa_offset = unwind_table->rows[table_idx].cfa_offset;
-
-    u64 frame_address = 0;
-
     bpf_printk("\tcfa reg: %d, offset: %d (pc: %llx)", found_cfa_reg, found_cfa_offset, found_pc);
+
+    u64 previous_rsp = 0;
     if (found_cfa_reg == X86_64_REGISTER_RBP) { 
-      frame_address = current_rbp + found_cfa_offset;
+      previous_rsp = current_rbp + found_cfa_offset;
     } else if (found_cfa_reg == X86_64_REGISTER_RSP) {
-      frame_address = current_rsp + found_cfa_offset;
+      previous_rsp = current_rsp + found_cfa_offset;
     } else {
       bpf_printk("\t[error] register %d not valid (expected $rbp or $rsp)", found_cfa_reg);
       return 0;
     }
-    
+    // TODO(javierhonduco): A possible check could be to see whether this value is within
+    // the stack. This check could be quite brittle though, so if we add it, it would be
+    // best to add it only during development.
+    if (previous_rsp == 0) {
+      bpf_printk("[error] previous_rsp should not be zero.");
+    }
+
+    // HACK(javierhonduco): We assume that the return address is *always* 8 bytes ahead of the previous stack
```

Comment:
I assume we'll use the value from the table before we merge everything to the main.

Reply:
I would first check how a varied set of real-life binaries store their return address and decide what to do

Reply:
Let's check the real-life data. But I'd do it reverse. So let's use the table, and then we can remove it.